.github/workflows/test.yml

```diff
@@ -88,3 +88,81 @@ jobs:
               xvfb-run python make.py check &&
               echo 'Finished successfully! :)'
             "
+
+  test-pios:
+    name: Test Raspbian ${{ matrix.docker-tag }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        docker-tag: ['stretch-2018-03-13', 'buster-2021-03-25']
+      fail-fast: false
+    services:
+      rpios:
+        # Custom made images for Mu: https://github.com/carlosperate/docker-qemu-rpi-os
+        image: ghcr.io/carlosperate/qemu-rpi-os-lite:${{ matrix.docker-tag }}-mu
+        ports: ["5022:5022"]
+    steps:
+      # This delay is a bit hacky, but can't find a way to signal when OS is ready
+      - name: Wait 2m30s for the docker image to start up QEMU and Raspberry Pi OS
+        run: sleep 150
+      - name: Clone project & setup it as the bash entry directory
+        uses: appleboy/ssh-action@master
+        with:
+          host: rpios
+          username: ${{ secrets.PI_USERNAME }}
+          password: ${{ secrets.PI_PASSWORD }}
+          port:  ${{ job.services.rpios.ports[5022] }}
+          script: |
+            mkdir ~/mu && cd ~/mu
+            git init
+            git remote add origin ${{ github.server_url }}/${{ github.repository }}.git
+            git fetch --progress --depth=1 origin ${{ github.sha }}
+            git checkout --progress FETCH_HEAD
```

Comment:
We cannot use the standard "git checkout" GH Action step, so we have to figure out what to checkout. I've based this on the official Action step source code, but I'm not 100% sure if it'll work, so I've created this PR from my fork to check it: https://github.com/mu-editor/mu/pull/1593/

Reply:
PR #1593 failed correctly, so this part is correct.

Basically GitHub creates a detached merge commit in this repository and it passes this hash to the CI, so these steps don't need to add the fork as a remote.